code/game/objects/items/pet_carrier.dm

```diff
@@ -0,0 +1,195 @@
+//Used to transport little animals without having to drag them across the station.
+//Comes with a handy lock to prevent them from running off.
+/obj/item/pet_carrier
+	name = "pet carrier"
+	desc = "A big white-and-blue pet carrier. Good for carrying <s>meat to the chef</s> cute animals around."
+	icon = 'icons/obj/pet_carrier.dmi'
+	icon_state = "pet_carrier_open"
+	item_state = "pet_carrier"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	force = 5
+	attack_verb = list("bashed", "carried")
+	w_class = WEIGHT_CLASS_BULKY
+	throw_speed = 2
+	throw_range = 3
+	materials = list(MAT_METAL = 7500, MAT_GLASS = 100)
+	var/open = TRUE
+	var/locked = FALSE
+	var/list/occupants = list()
+	var/occupant_weight = 0
+	var/max_occupants = 3 //Hard-cap so you can't have infinite mice or something in one carrier
+	var/max_occupant_weight = MOB_SIZE_SMALL //This is calculated from the mob sizes of occupants
+
+/obj/item/pet_carrier/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
```

Comment:
really, is this seriously what you're going to do to track occupants

Reply:
What would you rather I do? Check for nulls on removal and adding?

Reply:
Just use Exited. This absolute does not need to process anything.

Reply:
I wrote this code at like 1 AM, cut me some slack.